src/ray/util/periodical_runner.h

```diff
@@ -0,0 +1,44 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <boost/asio.hpp>
+#include <boost/asio/deadline_timer.hpp>
+
+namespace ray {
+
+/// \class PeriodicalRunner
+/// A periodical runner attached with an io_context.
+/// It can run functions with specified period. Each function is triggered by its timer.
+/// To run a function, call `RunFnPeriodically(fn, period_ms)`.
+class PeriodicalRunner {
+ public:
+  PeriodicalRunner(boost::asio::io_service &io_service);
+
+  ~PeriodicalRunner();
+
+  void RunFnPeriodically(std::function<void()> fn,
+                         boost::posix_time::milliseconds period);
```

Comment:
nit: this function can accept a `long`, so callers don't need to construct a posix_time before calling it.